lib/src/set_multimap/built_set_multimap.dart

```diff
@@ -0,0 +1,191 @@
+  ///
+  /// Wrong: `new BuiltSetMultimap({1: ['1'], 2: ['2'], 3: ['3']})`.
+  ///
+  /// Right: `new BuiltSetMultimap<int, String>({1: ['1'], 2: ['2'], 3: ['3']})`,
+  ///
+  /// Rejects nulls. Rejects keys and values of the wrong type.
+  factory BuiltSetMultimap([multimap = const {}]) {
+    if (multimap is BuiltSetMultimap<K, V>) {
+      return multimap;
+    } else if (multimap is Map ||
+        multimap is SetMultimap ||
+        multimap is BuiltSetMultimap) {
+      return new BuiltSetMultimap<K, V>._copyAndCheck(
+          multimap.keys, (k) => multimap[k]);
+    } else {
+      throw new ArgumentError(
+          'expected Map, SetMultimap or BuiltSetMultimap, got ${multimap.runtimeType}');
+    }
+  }
+
+  /// Creates a [SetMultimapBuilder], applies updates to it, and builds.
+  factory BuiltSetMultimap.build(updates(SetMultimapBuilder<K, V> builder)) =>
+      (new SetMultimapBuilder<K, V>()..update(updates)).build();
+
+  /// Converts to a [SetMultimapBuilder] for modification.
+  ///
+  /// The `BuiltSetMultimap` remains immutable and can continue to be used.
+  SetMultimapBuilder toBuilder() => new SetMultimapBuilder<K, V>(this);
+
+  /// Converts to a [SetMultimapBuilder], applies updates to it, and builds.
+  BuiltSetMultimap<K, V> rebuild(updates(SetMultimapBuilder<K, V> builder)) =>
+      (toBuilder()..update(updates)).build();
+
+  /// Converts to a [Map].
+  ///
+  /// Note that the implementation is efficient: it returns a copy-on-write
+  /// wrapper around the data from this `BuiltSetMultimap`. So, if no mutations are
+  /// made to the result, no copy is made.
+  ///
+  /// This allows efficient use of APIs that ask for a mutable collection
+  /// but don't actually mutate it.
+  Map<K, BuiltSet<V>> toMap() => new CopyOnWriteMap<K, BuiltSet<V>>(_map);
+
+  /// Deep hashCode.
+  ///
+  /// A `BuiltSetMultimap` is only equal to another `BuiltSetMultimap` with
+  /// equal key/values pairs in any order. Then, the `hashCode` is guaranteed
+  /// to be the same.
+  @override
+  int get hashCode {
+    if (_hashCode == null) {
+      _hashCode = hashObjects(_map.keys
+          .map((key) => hash2(key.hashCode, _map[key].hashCode))
+          .toList(growable: false)..sort());
+    }
+    return _hashCode;
+  }
+
+  /// Deep equality.
+  ///
+  /// A `BuiltSetMultimap` is only equal to another `BuiltSetMultimap` with
+  /// equal key/values pairs in any order.
+  @override
+  bool operator ==(other) {
+    if (other is! BuiltSetMultimap) return false;
+    if (other.length != length) return false;
+    if (other.hashCode != hashCode) return false;
+    for (final key in keys) {
+      if (other[key] != this[key]) return false;
+    }
+    return true;
+  }
+
+  String toString() => _map.toString();
+
+  // SetMultimap.
+
+  /// As [SetMultimap], but results are [BuiltSet]s and not mutable.
+  BuiltSet<V> operator [](K key) {
+    final result = _map[key];
+    return identical(result, null) ? _emptySet : result;
```

Comment:
== is a method call, identical is a built in. So it's clearer what's happening with identical. And probably faster, although I didn't check :)